db/migrate/20200921135723_create_host_facets_infrastructure_facets.rb

```diff
@@ -0,0 +1,13 @@
+class CreateHostFacetsInfrastructureFacets < ActiveRecord::Migration[6.0]
+  def change
+    create_table :host_facets_infrastructure_facets do |t|
+      t.references :host, type: :integer, foreign_key: true, index: { unique: true }
+      t.string :foreman_uuid
+
+      t.references :smart_proxy
+      t.string :smart_proxy_uuid
+
+      t.timestamps
```

Comment:
I would suggest to avoid timestamps for facts unless you really need this for some feature. Updating anything on every fact upload is a waste and generally we would like towards the opposite direction in the future for facts.

Reply:
Right, removed.